packages/vite/src/node/ssr/runtime/__tests__/server-worker-runner.spec.ts

```diff
@@ -1,8 +1,50 @@
 import { BroadcastChannel, Worker } from 'node:worker_threads'
 import { describe, expect, it, onTestFinished } from 'vitest'
-import { DevEnvironment, RemoteEnvironmentTransport } from '../../..'
+import type { HotChannel, HotChannelListener, HotPayload } from 'vite'
+import { DevEnvironment } from '../../..'
 import { createServer } from '../../../server'
 
+const createWorkerTransport = (w: Worker): HotChannel => {
+  const handlerToWorkerListener = new WeakMap<
+    HotChannelListener,
+    (value: HotPayload) => void
+  >()
+
+  return {
+    send: (data) => w.postMessage(data),
+    on: (event: string, handler: HotChannelListener) => {
+      if (event === 'connection') return
+
+      const listener = (value: any) => {
```

Comment:
Typing `value: HotPayload` can show the type error I mentioned above.

```suggestion
      const listener = (value: HotPayload) => {
```